docs/content/guides/dagster-pipes.mdx

```diff
@@ -0,0 +1,77 @@
+---
+title: Dagster Pipes | Dagster Docs
+description: "Dagster Pipes provides a protocol between the orchestration environment (Dagster) and external execution (ex: Databricks) and a toolkit for building implementations of that protocol."
+---
+
+# Dagster Pipes
+
+Dagster Pipes is a toolkit for building integrations between Dagster and external execution environments. It standardizes the process of passing parameters, injecting context information, ingesting logs, and collecting metadata all while remaining agnostic to how remote computations are launched in those environments. This enables the separation of orchestration and business logic in the Dagster ecosystem.
+
+It also smooths the process of incorporating pre-existing code, business logic, and execution environments into Dagster. Using Dagster Pipes, you add just a few lines to your code, and it becomes seamlessly invocable by the orchestrator. In turn, you can stream logs and metadata back to Dagster so you can leverage its observability, lineage, cataloging, and debugging capabilities.
+
+---
+
+## Benefits
```

Comment:
I think it would be helpful to clarify the answer to the question: "what is the best practice way for using Dagster?"

If the answer is more of a decision tree, we should include that decision tree

But I think it is very important that we clarify for both new dagster prospects and existing customers how pipes relates to assets w/out pipes (and maybe even just as important, our integrations w/out pipes)

In other words, a good addition would be: 
- when should I use pipes?
- when should I not?

Reply:
I think a cohesive decision tree about overall integration usage is beyond the scope of this guide given the muddled state of our integration ecosystem. @PedramNavid and I have discussed this in the past and I think making progress on this is a good candidate for quality corner.

In the short-term, I think we should frame Pipes as a much better way of richly integrating with external execution environments that cannot or do not include the core dagster module, and where Dagster does not control the deployment of code to those external environments.